TeamCode/src/main/java/org/firstinspires/ftc/teamcode/DriveTrain.java

```diff
@@ -0,0 +1,651 @@
+
+    //objects for controlling the rear two motors
+    private DcMotor leftRear;
+    //object for controlling the left rear motor
+    private DcMotor rightRear;
+
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// GYRO V
+    private AHRS gyro;
+    private AHRS.DimStateTracker gyroReset;
+    //gyro sensor
+
+    //variables for gyro sensor
+    private float goalDegrees;
+    // ???
+    private int goalPosition;
+    // ???
+    private int gyroRange;
+    // ???
+    private static final double     countsPerMotorRev    = 1440 ;
+    private static final double     driveGearReduction    = 1.5 ;
+    private static final double     wheelDiameterInches   = 4.0 ;
+    private static final double     countsPerInch         = (countsPerMotorRev * driveGearReduction) / (wheelDiameterInches * Math.PI);
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// SETUP V
+    private String LeftFront = "leftfront";
+    private String RightFront = "rightfront";
+    private String LeftRear = "leftrear";
+    private String RightRear = "rightrear";
+    private final int gyroPort = 3;
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// MISS V
+    private Telemetry telemetry;        //do we need this?
+    //object for reference (telemetry)
+    private boolean encodersCanRun;
+    private double goalEncoderPosition;
+    private double goalBackwardPosition;
+    private double goalLeftPosition;
+    private double goalRightPosition;
+    private double goalRightStrafePosition;
+    private double goalLeftStrafePosition;
+
+    private double testLiftPosition = 1;
+
+    private double currentGyro;
+
+    private double power;
+    /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// CONSTRUCT
+    //calls the second constructor of DriveTrain and passes a reference to the hardware map, telemetry, the 4 string names of the motors in the order left front, right front, left back, right back and the port reference to the gyro.
+    public DriveTrain(HardwareMap hardwareMap, Telemetry telemetry) {
```

Comment:
#Fix Can we pass in the `DcMotor`s directly instead of grabbing them through `HardwareMap`? That way you won't need to hardcode the motor config names in `DriveTrain` and it can be more easily moved to another robot with different motor config names.